src/connection.js

```diff
@@ -567,7 +570,8 @@ class Connection extends EventEmitter {
 
   clearRequestTimer() {
     if (this.requestTimer) {
-      return clearTimeout(this.requestTimer);
+      clearTimeout(this.requestTimer);
+      this.requestTimer = undefined;
```

Comment:
> We have a requestTimeout integration test in connection-test.coffee that tests the old behavior. It's still relevant, so we can keep it.
> To test the new behavior, we can add a test that'll sleep for a little over the requestTimeout amount in the first invocation of 'row' event handler. With old behavior that would cause a timeout, whereas with the new behavior it should not.

The existing test in connection-test.coffee uses the SQL statement `select 1 as C1;waitfor delay '00:00:05';select 2 as C2`. One would expect that the database server sends the first row immediately, waits for 5 seconds, and then sends the second row. But this is not the case. The first row is only received after the 1s request timeout has expired and the client has sent the ATTENTION message. While the data packet with the first row is received, the driver is in the SENT_ATTENTION state.